### PR TITLE
Add migration instructions for configuration files

### PR DESCRIPTION
**Pull request by kmakiela**
This PR introduces a first version of the migration guide. For now it includes changes in the configuration files only. 
Additionally, there are changes to the relative links to the other parts of the documentation as they were not working properly

### CODE REVIEW

File: guides/migrations/2.0.x_2.1.0.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Migration guide from 2.0.x to 2.1.0
```
Comment (michalwski) [inline guides/migrations/2.0.x_2.1.0.md:1]:
```suggestion
# Migration guide from 2.0.x to 2.1.0

```

File: guides/migrations/2.0.x_2.1.0.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+There are a few changes necessary in the configuration files and a few additional, optional ones.
+## Main config (`config/config.exs`)
```
Comment (michalwski) [inline guides/migrations/2.0.x_2.1.0.md:7]:
```suggestion

## Main config (`config/config.exs`)

```

File: guides/migrations/2.0.x_2.1.0.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+New logging entry may look like that:
+```elixir
+config :mongoose_push, :logging,
+  level: {:system, :atom, "PUSH_LOGLEVEL", :info},
+  format: {:system, :atom, "PUSH_LOGFORMAT", :json}
+```
+
+The structure changed a little and there is a new variable, `PUSH_LOGFORMAT`. It defaults to `:json` for the `prod` environment, and to `:logfmt` for the `dev` and `test` environments. The `logfmt` backend was changed to support structured logging. Please note that if you had a MongoosePush node configured with a log aggregation backend it probably needs to be reconfigured to match the current format of the log messages. You might also want to consider using the second standard, `JSON`, that MongoosePush now supports.
```
Comment (michalwski) [inline guides/migrations/2.0.x_2.1.0.md]: I find it more readable and easier to change later if the text in markdown files follow the rule "one sentence per line". Could you split this long line into several ones, each containing one sentence?

**Review by michalwski — COMMENTED** [review]
Thanks for the very comprehensive and detailed description of all the changes related to configuration. I left some comments on the code.
Also, I'm wondering in what case a MongoosePush user needs to change the elixir config files when switching to version `2.1.0`. In the most common case when MongoosePush is deployed from a docker image and configured with only env variables, there is no need to modify any of the `config/*.exs` files. Am I missing something?

File: guides/migrations/2.0.x_2.1.0.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      {MongoosePush.Config.Utils, :parse_bind_addr, []},
+      "PUSH_HTTPS_BIND_ADDR",
+      {127, 0, 0, 1}
+    },
+    port: {:system, :integer, "PUSH_HTTPS_PORT", 8443},
+    keyfile: {:system, :string, "PUSH_HTTPS_KEYFILE", "priv/ssl/fake_key.pem"},
+    certfile: {:system, :string, "PUSH_HTTPS_CERTFILE", "priv/ssl/fake_cert.pem"},
+    cacertfile: {:system, :string, "PUSH_HTTPS_CERTFILE", "priv/ssl/fake_cert.pem"},
+    protocol_options: [
+      # https://ninenines.eu/docs/en/cowboy/2.5/manual/cowboy_http/
```
Comment (aleklisi) [inline guides/migrations/2.0.x_2.1.0.md:78]: I am almost ready to bet that on cowboy update somebody will forget to update this link ;)

**Review by aleklisi — APPROVED** [review]
Thanks for the PR. it looks good.

File: guides/migrations/2.0.x_2.1.0.md (hunk 3 above)
Comment (jan-ciesla) [reply guides/migrations/2.0.x_2.1.0.md]: I've split the text into sentence per line in this paragraph and in the remaining ones.

**Review by jan-ciesla — APPROVED** [review]

**Comment by michalwski** [comment]
@janciesla8818 thanks for applying my suggestions. I think we still have one thing left to be sorted out:
> Also, I'm wondering in what case a MongoosePush user needs to change the elixir config files when switching to version 2.1.0. In the most common case when MongoosePush is deployed from a docker image and configured with only env variables, there is no need to modify any of the config/*.exs files. Am I missing something?

While it's good to have all the changes visible in the migration guide, I don't want to confuse people and rush them to change anything if it's not required.

**Comment by jan-ciesla** [comment]
@michalwski, thanks for the review comments. The first section of the document was updated to address this remark.

> Also, I'm wondering in what case a MongoosePush user needs to change the elixir config files when switching to version `2.1.0`. In the most common case when MongoosePush is deployed from a docker image and configured with only env variables, there is no need to modify any of the `config/*.exs` files. Am I missing something?

File: guides/migrations/2.0.x_2.1.0.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
+# Migration guide from 2.0.x to 2.1.0
+
+Please note that for standard usage, when configuring MongoosePush only with the environmental variables, the `config/*.exs` files are upgraded automatically and there is no action required from your end.
```
Comment (NelsonVides) [inline guides/migrations/2.0.x_2.1.0.md]:
```suggestion
Please note that for standard usage, when configuring MongoosePush only with environmental variables, the `config/*.exs` files are upgraded automatically and there is no action required from your end.
```

File: guides/migrations/2.0.x_2.1.0.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+If your MongoosePush is using modified `config/*.exs` files, please have a closer look at the changes presented below to see if you have to adjust your custom configuration.
+In such case, there could be a few changes necessary in the configuration files and a few additional, optional ones.
+
+## Main config (`config/config.exs`)
+
+MongoosePush supports structured logging in 2.1.0 release. To use that feature, there is a need to change logger configuration to use the new default.
```
Comment (NelsonVides) [inline guides/migrations/2.0.x_2.1.0.md]:
```suggestion
MongoosePush supports structured logging from the 2.1.0 release. To use that feature, there is a need to change the logger configuration to use the new default.
```

File: guides/migrations/2.0.x_2.1.0.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  ],
+  debug_errors: false,
+  code_reloader: false,
+  check_origin: true,
+  server: true
+  ```
+  As you can see, there is no change in the variables used to parse the options.
+
+### Logging
+Old logging entry looked like that:
```
Comment (NelsonVides) [inline guides/migrations/2.0.x_2.1.0.md]:
```suggestion
Old logging entry looked like this:
```

File: guides/migrations/2.0.x_2.1.0.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+```elixir
+config :mongoose_push, loglevel: {:system, :atom, "PUSH_LOGLEVEL", :info}
+```
+
+New logging entry may look like that:
```
Comment (NelsonVides) [inline guides/migrations/2.0.x_2.1.0.md]:
```suggestion
New logging entry may look like the following:
```

**Review by NelsonVides — COMMENTED** [review]
Very nice!
Just have one question, shouldn't we mention the deprecation of API v2? @michalwski?
Otherwise it's very ready to merge (except for some usual articles) 😄

**Review by NelsonVides — APPROVED** [review]
💯

**Pull request merged**